Notifications: Fix Slideshow block images

ivan-ottinger opened the pull request:
#### Changes proposed in this Pull Request

The Slideshow block images are not displayed in the post notifications:
https://github.com/Automattic/wp-calypso/issues/46964

The proposed code address this issue by overriding the Slideshow block CSS.

We also make sure the notification displays a maximum of four Slideshow images.

This is due to limited notification space.

#### Testing instructions

1. Make sure there is a subscriber on your site that can receive notifications on newly published posts.

Notification settings are located in the subscriber's account at `/following/manage` → ⚙️ Settings → "Notify me of new posts" (toggle):

![Markup_on_2021-07-29_at_15_35_56](https://user-images.githubusercontent.com/25105483/127670085-0ca7a053-6f1f-4122-9610-9f8692f1c5ad.png)

2. Log in to the account of the user that can publish posts on the site.
3. Create and publish a new post that has the Slideshow block with multiple images in it.
4. Review the notification in the subscriber's account. The Slideshow images should display:

![Markup on 2021-07-30 at 16:45:13](https://user-images.githubusercontent.com/25105483/127670341-83f3640f-af19-4bc2-a5fb-0c7b90e5f3af.png)

Please note that the proposed change is work in progress.

ivan-ottinger committed:
Fix Slideshow images display in Notifications

The Slideshow block images are not displayed in the post notifications:

https://github.com/Automattic/wp-calypso/issues/46964.

The proposed code fixes this issue by overriding the Slideshow CSS.

We also make sure the notification displays a maximum of four images.

This is due to limited notificaion space.

ivan-ottinger committed:
Fix linter error

This commit moves the definition of slideCount outside from `switch/case` statement.

yansern commented on apps/notifications/src/panel/templates/body.jsx:
```diff
@@ -114,6 +115,31 @@ export class NoteBody extends React.Component {
 				case 'post':
+					slideCount = 0;
+					for ( i = 0; i < block.block.ranges.length; i++ ) {
+						if ( block.block.ranges[ i ].type === 'figcaption' ) {
```
Perhaps we can assign `block.block.ranges[ i ]` to a variable.

ivan-ottinger replied:
Thank you for suggestion Yan! I am now addressing that in the following commit: https://github.com/Automattic/wp-calypso/pull/55062/commits/3e456bd3f42de201bff1f5dbbe5e65babad2db63.

ivan-ottinger committed:
Move block adjustments to a dedicated function

The commit moves the block adjustments to a separate function in functions.jsx.

This improves the readability of the code.

It will also help us expand the block adjustments moving forward.

We will be able to address formatting issues that come up with other blocks.

ivan-ottinger committed:
Assign `block.ranges[ i ]` to a variable `range`

The commit assigns `block.ranges[ i ]` to a variable `range`.

This makes the code more readable.

ivan-ottinger commented:
We are now looking for a better way of addressing the issue in D65094-code - with the help of block rendering fallback.

ivan-ottinger commented:
Closing in favor of D65094-code.
